### DISCUSSION TITLE
Data Annotations não funciona

### DISCUSSION THREAD

**Discussion by axpalx** [balta-io/2811#29] [Q&A] [answered]
Olá, estou fazendo algo errado e não encontro o que pode ser. Os Data Annotations REQUIRED não estão funcionando.
Quando eu tento gravar uma categoria nova os Data Annotations não funionam e caio no primeiro Catch("Não foi possível retornar a categoria").
Só funciona fazendo a validação no método.

```
            if (!ModelState.IsValid)
                return BadRequest(ModelState);
```

Obrigado

Segue minha classe VIEWMODELS
```
using System.ComponentModel.DataAnnotations;

namespace Blog.ViewModels
{
    public class EditorCategoryViewModel
    {
        [Required]
        public string Name { get; set; }

        [Required]
        public string Slug { get; set; }
    }
}
```
Meu método POST.
```
[HttpPost("v1/categories/")]
        public async Task<IActionResult> PostAsync(
            [FromBody] EditorCategoryViewModel model,
            [FromServices] BlogDataContext context)
        {
            try
            {
                var category = new Category
                {
                    Id = 0,
                    Name = model.Name,
                    Slug = model.Slug.ToLower(),
                };

                await context.Categories.AddAsync(category);
                await context.SaveChangesAsync();

                return Created($"v1/categories/{category.Id}", category);
            }
            catch (DbUpdateException ex) { 
                return StatusCode(500, "Não foi possível retornar a categoria");
            }
            catch (Exception ex)
            {
                return StatusCode(500, "Falha interna no servidor");
            }
        }
```
Meu retorno no Postman após tentar gravar uma nova categoria com os campos Name e Slug em branco.
![image](https://user-images.githubusercontent.com/16595443/184121001-d2966be6-b63f-499d-8408-cac49758b527.png)

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia @axpalx, para as validações funcionarem de forma automática você precisa herdar do `ApiController`, caso contrário você pode executar uma validação `ModelState.IsValid` para saber se ocorreu algum erro.

**Reply by axpalx** [reply]
Bom dia @andrebaltieri!
Eu só entendi a sua explicação após assistir o video da aula "Padronizando erros".
Agora está tudo funcionando, muito obrigado.

**Reply by andrebaltieri** [reply]
💜
